From d82766108d1d25f97b1f73baadb56e92a581e822 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Wed, 9 Oct 2019 11:05:00 -0500
Subject: [PATCH] 1635-os_path_join

hub: check if path is absolute in get_upload_path

Original by Yuming Zhu <yzhu@redhat.com>
from https://pagure.io/koji/pull-request/1635
---
 hub/kojihub.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hub/kojihub.py b/hub/kojihub.py
index 0b222c9..14355d2 100644
--- a/hub/kojihub.py
+++ b/hub/kojihub.py
@@ -12526,7 +12526,7 @@ def get_upload_path(reldir, name, create=False, volume=None):
     if d or name.startswith('.'):
         raise koji.GenericError("Invalid upload filename: %s" % orig_name)
     reldir = os.path.normpath(reldir)
-    if not reldir or reldir.startswith('..'):
+    if not reldir or reldir.startswith('..') or os.path.isabs(reldir):
         raise koji.GenericError("Invalid upload directory: %s" % orig_reldir)
     if volume is not None:
         # make sure the volume is valid
-- 
2.6.3

